Skyculture "Ruanui sky : Tahiti and Society islands"

ThibaudGS opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

### Description
<!--- Please include a summary of the change and which issue is fixed. -->
<!--- Please also include relevant motivation and context. -->
<!--- List any dependencies that are required for this change. -->

A new skyculture was added "Ruanui sky : Tahitit and Society islands".
This skyculture contains stars and constellations used in traditional polynesian astronomy in Society Islands (Tahiti, Bora bora...).
The main book reference we used was "Ancient Tahiti", therefore these stars are mainly the stars taught by Ruanui, an old lady from Bora bora whose knowledge was translated in the Book "Ancient Tahiti" by Teuira Henry.

This skyculture was missing, and our working group is proud to add this relevant addition to the Stellarium software, so that the polynesian population, but also a larger audience, can access this traditional knowledge.

The working group was composed of tahitian experts in polynesian astronomy (Christine Briant, Libor Prokop, Regis Plichart) and cultural tahitian associations ("Fa'afaite i te ao maohi", "Te Fare Hiroa no Vavau").

Fixes # (issue)

### Screenshots (if appropriate):
![Maui's hook](https://github.com/user-attachments/assets/44683ca3-a5bf-4dd6-9f52-ecb0eacd78b5)
![Stellarium2](https://github.com/user-attachments/assets/4f3c4adb-facb-485f-8468-d99e82d23c47)

### Type of change
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] This change requires a documentation update
- [ ] Housekeeping

### How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

This code has been tested on a computer version of Stellarium 24.1
We tested : star names, constellations names, constellation ships, constellation illustrations (with seasonal rules), planets names

**Test Configuration**:
* Operating system: <Name, version number>
Windows 11

* Graphics Card: <Manufacturer (likely Intel, NVidia, AMD?), Model (HD, Geforce, Radeon..., with model number), driver version?>
Intel(R) Core(TM) i7-1065G7 CPU @ 1.30GHz   1.50 GHz
NVidia GeForce 

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the [code style](http://stellarium.org/doc/head/codingStyle.html) of this project.
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (header file)
- [ ] I have updated the respective chapter in the Stellarium User Guide
- [ ] My changes generate no new warnings

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

alex-w commented:
Since version 25.1 we have a new format for storing the data of sky cultures and conversion tool from old format to the new one.

At the moment I have a few questions:
* Having the people of Tahiti and Society islands specific writing system or it uses a latin alphabet?
* Could you share an English names for planets also?
* Could you share an English names for stars also?
* What meaning names like "A5/T5" or "A2b" - is it really names or just some designations?

ThibaudGS commented:
Hello Alex-w,

Thank you for your reply !

Where can I find the conversion tool ?

Here are the anwsers :
- The people of Tahiti use a latin alphabet with some special letters like "ā" and the use of apostrophe
- I could share a translation for the planet and star names in english, but where do I put it ? Is it in a new file ? I choose the Hawaian skyculture as an example, and I can't find the translations for the stars and planets.
- "A5/T5" and "A2b" are some designations used to distinguish some stars that may have different names depending on the reference used.

Thank you for your help !

gzotti commented:
Please see User Guide chapter 9.

ThibaudGS commented:
I updated the file for the planets names, using the user guide. For instance Saturn look now like this :
Saturn	"Fetūtea"	_("Pale star")

But I fail to find a way to do the same with stars. According to the user guide, the code for a star should look like this : 
102098|_("Ta'urua-i-te-ha'apāra'a-manu") 1

Where can I put the english translation for the native name of a star ? I agree it would be very interesting as the translation is often a complete sentence.

Thanks a lot for your help !

Thibaud

gzotti commented:
Please update to current version 25.1 and then use the User Guide that comes with version 25.1, chapter 9, to find all instructions, and take skycultures that come with 25.1 for examples. Convert your work with the converter that comes with version 25.1, then continue work on the converted result. Note that 25.1 does not show all cultural names on screen yet, but 25.2 (scheduled for June) will.

ThibaudGS commented:
Thank you qzotti, i'm working on it

sushoff commented:
better examples are the Siberian, Chinese and Greek (Almagest) - I think, they are the most complete (in terms of foreign characters, translations etc.) 

Good luck!

gzotti commented:
@ThibaudGS any updates? Are you planning completion ready for 25.2 or later? To see most new settings affecting name labelling, use a https://github.com/Stellarium/stellarium-data/releases/tag/weekly-snapshot

ThibaudGS commented:
Hi ! I finished the corrections ! Both files "index.json" and "description.md" are working on my computer (Stellarium 25.1). Both illustrations are in the "illustrations" folder, and it's working.
I didn't delete the old files, because we can still see them in Samoan ou Chinese skyculture, which I used as esamples. But I guess theses old files are useless now.
I hoe i'm on time for the 25.2 :)

ThibaudGS commented:
Ok, thanks. I wait for the instructions.

ThibaudGS commented:
I tried to understand how to rebased the branch, but it is my first time in Github and I don't understand...
Is it something I have to do on my Windows system with the Windows command ? Or is-it something I can do on this website ?
My apologies for my poor level in Github.
Wouldn't it be easier for me to do it again from the beginning, based on the 25.1 ?

10110111 commented:
> Sorry, I click on "update with rebase" on this page... a mistake.

Why? Looks like this branch is successfully rebased.

> I haven't made these modificications

You should, otherwise your SC isn't properly integrated into the build system.

ThibaudGS commented:
Thanks !
I made the modifications.

I had the "ADD_SUBDIRECTORY" in  skycultures/CMakeLists.txt like this : 
"...
ADD_SUBDIRECTORY( norse )
ADD_SUBDIRECTORY( northern_andes )
ADD_SUBDIRECTORY( romanian )
ADD_SUBDIRECTORY( ruanui_sky_tahiti_and_society_islands )
ADD_SUBDIRECTORY( russian_siberian )
ADD_SUBDIRECTORY( sami )
ADD_SUBDIRECTORY( samoan )
..."

I modified the [CMakeLists.txt] file (https://github.com/ThibaudGS/stellarium/blob/master/skycultures/ruanui_sky_tahiti_and_society_islands/CMakeLists.txt), like this : 

get_filename_component(/skycultures/ruanui_sky_tahiti_and_society_islands "${CMAKE_CURRENT_SOURCE_DIR}" REALPATH)
get_filename_component(ruanui_sky_tahiti_and_society_islands ${/skycultures/ruanui_sky_tahiti_and_society_islands} NAME)

install(DIRECTORY ./ DESTINATION ${SDATALOC}/skycultures/${ruanui_sky_tahiti_and_society_islands}
        FILES_MATCHING PATTERN "*"
        PATTERN "CMakeLists.txt" EXCLUDE)

I'm not sure if this last part is done well.

If everynthing is OK, do I have to do something else now ? Or the final "push" action is on your side ?

Thank you again for your help and patience.

gzotti commented on skycultures/ruanui_sky_tahiti_and_society_islands/index.json:
```diff
@@ -0,0 +1,203 @@
+  ],
+ 
+  "common_names": {
+  "HIP 24608": [{"english": "Tahi-ari'i", "references": [1]}],
```
These star names should all be tagged as "native". If they have a meaning (something like "red star with two companions") a translation should be given in "english".

gzotti commented on skycultures/ruanui_sky_tahiti_and_society_islands/index.json:
```diff
@@ -0,0 +1,203 @@
+"NAME Jupiter": [{"english": "Great festivity who struck the zenith of the sky", "native": "Ta'urua-nui-i-tu'i-i-te-porou-o-te-ra'i"}],
+"NAME Mars": [{"english": "Fading redness", "native": "Maunu-'ura"}],
+"NAME Mercury": [{"english": "Mercury", "native": "Ta'ero"}],
+"NAME Moon": [{"english": " Moon ", "native": "Marama"}],
```
The spaces surrounding "Moon" in the english name should be omitted.

ThibaudGS replied:
i've made the modification

gzotti commented:
You can and indeed should delete now all *.fab and *.utf8 files, and info.ini, All relevant information is in index.json and description.md. I see a few typos in description.md, but it may be easier to fix them after merge. We have left a few fab files in other skycultures, but only in the repo, in case the author has vanished but the file contains something important that was not integrated.

ThibaudGS commented:
Ok, I deleted the files

Atque commented:
There's a small typo in one of the constellations.

Row 70 of `index.json`
      "common_name": {"english": "Vortex ocean in wich to lose crime", "native": "Moana-'ohu-noa-'ei-ha'amoe-hara"}

"Vortex ocean in wich to lose crime"
should be
"Vortex ocean in which to lose crime" (missing an H in _which_).

gzotti commented:
Yes, there are a few such minor issues. Instead of describing each in long sentences, I can fix them in a post-merge cleanup. Or if you want, you can send a pull request. 

@ThibaudGS I wonder, are the figures of Orion (with shield and raised club), Gemini, Auriga, Perseus,  Hydra, Crater, Corvus, Pisces, Cygnus really seen like "in the West", just with different names (which mostly don't seem to fit to those figures)? It looks strange to me. Other than that, technically it's OK now for me. @alex-w ? @sushoff ?

alex-w commented:
Western stick figures for the local constellations looks strange for me too. Is it western influence or unknown stick figures for these constellations?

alex-w commented on skycultures/ruanui_sky_tahiti_and_society_islands/description.md:
```diff
@@ -0,0 +1,49 @@
+# Ruanui sky : Tahiti and Society islands
```
Please remove extra white space before “:”

gzotti replied:
These were the small typos mentioned above. Quick to fix for us after squash-merging as well.

alex-w commented on skycultures/ruanui_sky_tahiti_and_society_islands/description.md:
```diff
@@ -0,0 +1,49 @@
+
+## Introduction
+
+French Polynesia encompasses 118 islands scattered amidst a marine area of nearly 5 millions km2. It is divided into 5 archipelagos : the Society archipelago , the Austral , the Tuamotu, the Gambier and the Marquesas archipelago. The Society archipelago, composed of 6 islands and 8 atolls, is divided into 2 subgroups, the Windward islands and the Leeward islands. This archipelago has developed it ...
```
Please remove extra white space before “:” and `km2` should be rephrased (`km<sup>2</sup>` or `square kilometers`)

alex-w commented on skycultures/ruanui_sky_tahiti_and_society_islands/description.md:
```diff
@@ -0,0 +1,49 @@
+
+### Linking space, time and men : an "environnemental" calendar
+
+Based on very precise observations, the experts of ancient times or tahu'a developed an acute knowledge of the stellar movement, linking closely the sky, the space and the time to the cycles of the environment, of the animals and of the plants. They were indissociable. This knowledge was passed down in the form of environmental calendars. Known and controlled by the elites, leaders and specialist ...
```
Please remove extra white spaces

alex-w commented on skycultures/ruanui_sky_tahiti_and_society_islands/description.md:
```diff
@@ -0,0 +1,49 @@
+
+"Star navigation" was the subject of many research in order to determine how the reading of the sky allowed the Polynesian people to navigate with such accuracy during journeys over thousands of km between remote islands. Today the names of the stars and of the constellations are becoming keys to approach the ancient voyagers' vision of the sky : markers to reach a target-island and to come back  ...
+
+Today, the "Fa'afaite i te Ao Ma'ohi" association keeps these traditional navigation techniques alive by travelling between the polynesian islands on a traditional Polynesian double-hulled canoe.</p>
```
No need `</p>` here

ThibaudGS commented:
I corrected the Typo "wich" in index.
I added the two missing english translations on constellations.
About the stick figures, the explanation is that the main source for the tahitian constellations is the book "Ancient tahiti", Teuira Henry, 1928. In this book, a direct link is made beetween the western constellations and the tahitian ones, for the names. But there is no information about the traditionnal "form" tahitian people might have seen in these constellations. Our working group decided to keep the western stick figure, as it is the closest choice to respect the source "Ancient Tahiti".
If a consensus arise in the future about these figures among Tahiti traditionnal astronomers, we will make the update on the skyculture.

ThibaudGS commented:
The only consensus find to diverge from western figures was : "Hiro's kite" and "Maui's hook" (for the Scorpio) and the "Cluster of four" (for the Crux)

alex-w commented:
So, OK for me, but with typofixes after merge...

ThibaudGS commented:
I changed  slightly the stick figures in index file. Mostly to make it different from the western visual, in a very minimalist way. A more elaborate work would need discussion with my working group which may take a long time. The next step for this skyculture will be to find illustrations for all the constellations, with the help of more people from tahitian community. I'm sure that the stick figures will be more approprietly modified when this (exciting !) work will be done, for a future extension of Stellarium.

ThibaudGS commented:
Thank you !! Mauruuru roa ! Now that the branch is merged, do I have something to do ?

ThibaudGS commented:
Hi !
For the traduction in French you can use the traduction from the french version of the book "Ancient Tahiti"
Here are the traductions for the constellations :

001 "Matari'i" _("Les petits yeux")
002 "Tau-hā" _("Les quatre liés")
003 "Matau-a-maui" _("Le hameçon de Maui")
004 "Fa'a-nui" _("La grande vallée")
005 "Fa'a-iti" _("La petite vallée")
007 "Atu-tahi" _("La première bonite")
008 "Atu-ra'i-putuputu" _("Banc de bonites célestes")
010 "Pira'e-tea" _("La sterne blanche")
011 "Metua-'ai-papa" _("Parent mangeur de roc")
012 "Nā-hui-tārava-ia-mere" _("Les étoiles alignées de Mere")
013 "Moana-'ohu-noa-'ei-ha'amoe-hara" _("Océan tourbillonant pour faire disparaître les fautes")
014 "Nā-'ainanu" _("Les deux qui désirent la nourriture")
015 "Fa'a-tā-potupotu" _("La vallée ouverte")
016 "Nā-matarua" _("Les deux yeux")
017 "Uru-meremere" _("Forêt de tendresse des parents")
018 "Te-'uo-a-hiro" _("Le cerf-volant de Hiro")
019 "Rua-o-mere" _("Caverne de la tendresse des parents")
020 "Tū-i-te-moana-urifā" _("Qui se tient dans l'océan à l'odeur nauséabonde")
021 "Pipiri-ma" _("Les avares")

For most of them, it is unclear what it really means, which may be challenging to translate. For "Parent eater of rock", it literally translates in french "Parents mangeur de roc" (the parents (father and/or mother) who eat stone). 

I think we should use these french translations from "Ancient Tahiti", which are the more accepted among the community.

gzotti commented:
Are living Tahitians involved in this project? They should know what the stone eater is. There is a fish that bears this name, at least in some languages. Or is it a folk tale about ancient humans or ancestors who ate stones? (like "First settlers, they had nothing so they had to eat stones, and they started cultivating the islands...") Without context, this is not translatable and simply incomplete. 

"Atu-tahi" _("La première bonite") is translated to "Single bonito". Isn't it "First Bonito", and Bonito is a kind of Tuna? And then "Atu-ra'i-putuputu" _("Banc de bonites célestes") should be swarm of celestial Tuna, not numerous (mixed) fishes? 
I see "mere" in both entries around "parental yearnings" (inviting its interpretation as "parents"), and in "aligned stars of Mere" (untranslated), Would that be the "aligned stars of the parents"? 
Our translators start with the English version, and you could help them by providing further contextual information in the "translators_comments" JSON tag. Hopefully at some point before 25.3 please.

Are the numerical abbreviations indices to a list in one of the sources? If not, we recommend not to use numbers but memorizable labels of 2-4 characters that much better help in identification. They can be displayed on screen, but I find numbers as figure labels uninspiring. Something like FaN, FaI, FaP for the three valleys. Is there a story concerning these celestial valleys? Etc., etc. In case the old books are available online, the reference list may contain URLs.

ThibaudGS commented:
Yes, living tahitians are involved in this project.

The meaning of some of the constellations is lost. For instance, maybe "Parent eater of rock" is related to a fish, maybe to something else. We do not know.

After the arrival of european people in french polynesia during XVIIIe century, a lot of the traditonnal culture (an oral tradition) was erased. There is very few sources which we can relate on. The book "Ancient Tahiti" is one of them. It was written by Teuira Henry with the notes of her grand-father John M. Orsmond (1784-1856). The most important part for astronomers is the "Genealogy of the stars" which was told to John M. Orsmond by an old lady from Bora Bora called Ruanui.

In the Genealogy of the stars, we learn the names of the stars and constellations. Teuira Henry translated herself the tahitan names in english. Sometimes the translation seems innacurate or odd, but our working group has chosen to respect this traduction.

For some of the constellations, like "Maui's hook" or "Hiro's kite" some legends are known that I can add.

There also an order in which the constellations are "born" (like a "Genealogy") in the sky, wich can add some context.

Our working group can work on a list of the "context" we have (or not) for each constellation.

What is the deadline for this work, if it has to be done before 25.3 ? Is there a date scheduled for the release of 25.3 ?

Also, does it mean that the skyculture won't be available in 25.2 ?

gzotti commented:
The skyculture is in 25.2 as it is now, but can be enhanced whenever you want to provide new information for our users. We have releases every begin of seasons, so 25.3 is expected around Sept.23, earlier contribution means more time for our translators.  

Just my personal opinion is that confusing, incomplete or unclear descriptions lead to disappointment and turning away. My expectation was that the Tahitian word for this "parent eater of rock" is used in everyday language and is easily understood by Tahitians. I can only guess what it might mean, and there is a wide spectrum of possibilities. If this is not a name of a local fish found on the market, then so be it. If it is not a character from a well-known Tahitian folk tale that every child there knows, but even locals have no clue what it means, I will regretfully accept it and translate as inexplicable as this term is. Is the word that has been translated into "Parent" used for human and animal parent in the same way? Does that give any clue?

sushoff commented:
I recommend to add all the constellations with detailed description of etymology, sources, mythology ... to the [All Skies Encaclopaedia](https://xing.fmi.uni-jena.de/mediawiki/index.php/All_Skies_Encyclopaedia). from there, it will be readable inside and outside Stellarium. 

Please contact me by mail to get an account (susanne.hoffmann@uni-jena.de)

ThibaudGS commented:
Thank you, I understand the necessity to add these informations when available. I'll add context to the constellations with the help of our working group.

For the example of "Parent eater of rock", "Metua-'ai-papa". "Metua" means "Parents". Papa has different meanings. In the context of the genealogy of the stars, It can mean "earth", "primordial rock" or "rock". "'ai" might be a short word for "eat".
If the full name was a way to describe a fish, I guess Teuira Henry would have translated it differently.

I think it will be also useful for the user to know the concept of genealogy, and the "order" of birth for the constellations.

ThibaudGS commented:
Hello !

Thank you for adding the new skyculture to Stellarium 25.2.
I started to work on star names translations in english.

I have a few questions :
- Do I have to create a new branch and do a new pull request for the 25.3 ? I understand I didn't do it correctly the first time. Any advice on how to do it correctly ?
- I have the description text in french. Is it useful for the translators ? Where can I put it ?
- When I download the 25.2 on my computer, I can access the new skyculture. But not on my phone (mobile version, android). Is it normal ?

Thank you for all your help !

alex-w commented:
> I have a few questions :
> 
> * Do I have to create a new branch and do a new pull request for the 25.3 ? I understand I didn't do it correctly the first time. Any advice on how to do it correctly ?

Yes, each feature or serious changes via separate branch/PR

> * I have the description text in french. Is it useful for the translators ? Where can I put it ?

Please use Transifex to translate descriptions and other SC stuff

> * When I download the 25.2 on my computer, I can access the new skyculture. But not on my phone (mobile version, android). Is it normal ?

Yes

ThibaudGS commented:
Thank you for your anwsers.

I'll continue the work to improve this version of the skyculture, and I'll try to use Transifex.

Can you explain what is missing for this skyculture to be used on mobile version of Stellarium ? Is-it a question of time, or is-it missing files ?

10110111 commented:
> Can you explain what is missing for this skyculture to be used on mobile version of Stellarium ? Is-it a question of time, or is-it missing files ?

The projects are completely separate, so you need to ask @xalioth whether and when this SC can be included (or maybe write to Stellarium Mobile's support).

ThibaudGS commented:
Mauruuru roa ! Thank you ! I try to contact him :)
